### PR TITLE
Fixed exception caused by publishers being None

### PR DESCRIPTION
**Pull request by tedsta**

### CODE REVIEW

File: perfkitbenchmarker/publisher.py [hunk 1]
```diff
@@ -776,7 +776,7 @@ class SampleCollector(object):
       the run directory to the publishers list.
     run_uri: A unique tag for the run.
   """
-  def __init__(self, metadata_providers=None, publishers=None,
+  def __init__(self, metadata_providers=None, publishers=[],
```
Comment (ehankland) [inline perfkitbenchmarker/publisher.py]: I'd prefer checking if publishers is none below and initializing it to an empty list - I like staying away from mutable default args.

Also, I notice that publishers (and not self.publishers) is getting extended online 790 and 792 - is that what's intended?

Comment (tedsta) [inline perfkitbenchmarker/publisher.py]: Done

Good good catch, I believe it's supposed to modify self.publishers

File: perfkitbenchmarker/publisher.py [hunk 2]
```diff
@@ -785,11 +785,14 @@ def __init__(self, metadata_providers=None, publishers=None,
     else:
       self.metadata_providers = DEFAULT_METADATA_PROVIDERS
 
-    self.publishers = publishers[:]
+    if publishers is not None:
+      self.publishers = publishers[:]
+    else:
+      self.publishers = []
```
Comment (ehankland) [inline perfkitbenchmarker/publisher.py]: ```self.publishers = publishers[:] if publishers else []```
is more compact, but up to you

Comment (tedsta) [inline perfkitbenchmarker/publisher.py]: Cool, I like it. Done

**Pull request merged**